main/src/cgeo/geocaching/location/UTMConverter.java

```diff
@@ -4,44 +4,42 @@
  */
 package cgeo.geocaching.location;
 
-import com.bbn.openmap.proj.Ellipsoid;
-import com.bbn.openmap.proj.coords.LatLonPoint;
 import com.bbn.openmap.proj.coords.MGRSPoint;
 
 /**
- * Uses https://github.com/openmap-java/openmap to convert from latitude, longitude to UTM
+ * Uses https://github.com/openmap-java/openmap to convert between latitude/longitude and UTM
  */
 public class UTMConverter {
 
-/* parsing of UTM Strings is not possible with openmap lib, it depends on java.awt.geom.Point2D which is not available on Android
     public Geopoint utm2LatLon(final String utmString) {
         final MGRSPoint mgrsPoint = new MGRSPoint(utmString);
-        final LatLonPoint.Double latLonPoint = (LatLonPoint.Double) mgrsPoint.toLatLonPoint();
-        return new Geopoint(latLonPoint.getLatitude(), latLonPoint.getLongitude());
+        final double[] latlon = mgrsPoint.toLatLon();
+        return new Geopoint(latlon[0], latlon[1]);
     }
-*/
 
     public UTM latLon2UTM(final double latitude, final double longitude) {
-        final MGRSPoint utm = new MGRSPoint(new LatLonPoint.Double(latitude, longitude), Ellipsoid.WGS_84);
-        return new UTM(utm.easting, utm.northing, utm.zone_number, String.valueOf(utm.zone_letter));
+        final MGRSPoint utm = MGRSPoint.LLtoMGRS(new double[]{latitude, longitude});
+        return new UTM(utm.easting, utm.northing, utm.zone_number, String.valueOf(utm.zone_letter), utm.getMGRS());
     }
 
     public static class UTM {
         private final double easting;
         private final double northing;
         private final int longZone;
         private final String latZone;
+        private final String mgrs;
 
-        public UTM(final double easting, final double northing, final int longZone, final String latZone) {
+        public UTM(final double easting, final double northing, final int longZone, final String latZone, final String mgrs) {
             this.easting = easting;
             this.northing = northing;
             this.longZone = longZone;
             this.latZone = latZone;
+            this.mgrs = mgrs;
         }
 
         @Override
         public String toString() {
-            return longZone + latZone + " " + Math.round(easting) + " "
+            return longZone + latZone + " E " + Math.round(easting) + " N "
```

Comment:
Geocaching.com is showing them. But of course that doesn't mean it should be done that way  ;)